### PR TITLE
Remove optional height indexing interface

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

All VMs now implement height indexing - so we can add the methods to the VM interface.

## How this works

Moves methods from the `HeightIndexedChainVM` interface to the `ChainVM interface.

## How this was tested

CI

### CODE REVIEW

File: vms/proposervm/vm.go [hunk 1]
```diff
@@ -414,7 +405,7 @@ func (vm *VM) repair(ctx context.Context) error {
 		}
 	case block.ErrIndexIncomplete:
 	default:
-		return nil
+		return err
```
Comment (StephenButtolph) [inline vms/proposervm/vm.go:408]: @abi87 can you look into this? Seems like this should have been returning an error before

Comment (StephenButtolph) [reply vms/proposervm/vm.go:408]: May also want to add a regression test.

Comment (abi87) [reply vms/proposervm/vm.go:408]: @StephenButtolph  see [here](https://github.com/ava-labs/avalanchego/pull/1915)

File: proto/vm/vm.proto [hunk 2]
```diff
@@ -108,9 +108,8 @@ enum Error {
   ERROR_UNSPECIFIED = 0;
   ERROR_CLOSED = 1;
   ERROR_NOT_FOUND = 2;
-  ERROR_HEIGHT_INDEX_NOT_IMPLEMENTED = 3;
-  ERROR_HEIGHT_INDEX_INCOMPLETE = 4;
-  ERROR_STATE_SYNC_NOT_IMPLEMENTED = 5;
+  ERROR_HEIGHT_INDEX_INCOMPLETE = 3;
+  ERROR_STATE_SYNC_NOT_IMPLEMENTED = 4;
```
[the remark this answers is not in the document]
Comment (abi87) [reply proto/vm/vm.proto:112]: why the flipped values?

File: snow/engine/snowman/block/vm.go [hunk 3]
```diff
@@ -5,12 +5,19 @@ package block
...
 
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
 	"github.com/ava-labs/avalanchego/snow/engine/common"
 )
 
+// ErrIndexIncomplete is used to indicate that the VM is currently repairing its
+// index.
+//
+// TODO: Remove after v1.11.x activates.
```
Comment (abi87) [inline snow/engine/snowman/block/vm.go:18]: just reasoning aloud here: is this true even for the ProposerVM? I mean once we're sure that any VM will implement its height index, there won't be cases there proposerVM height index can be incomplete (following a successful initialization)?

**Review by abi87 — APPROVED** [review]
LGTM, apart from a couple of questions.

File: proto/vm/vm.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply proto/vm/vm.proto:112]: We always just bump the rpcchainvm version anyways - so no need for compatibility at the proto level.

File: snow/engine/snowman/block/vm.go (hunk 3 above)
Comment (StephenButtolph) [reply snow/engine/snowman/block/vm.go:18]: The index can still be repaired async until `v1.11.x`

**Review by darioush — APPROVED** [review]

**Pull request merged**